src/sagemaker/session.py

```diff
@@ -780,6 +780,31 @@ def delete_endpoint(self, endpoint_name):
         LOGGER.info('Deleting endpoint with name: {}'.format(endpoint_name))
         self.sagemaker_client.delete_endpoint(EndpointName=endpoint_name)
 
+    def delete_endpoint_config(self, endpoint_config_name):
+        """Delete an Amazon SageMaker endpoint configuration.
+
+        Args:
+            endpoint_config_name (str): Name of the Amazon SageMaker endpoint configuration to delete.
+        """
+        LOGGER.info('Deleting endpoint configuration with name: {}'.format(endpoint_config_name))
+        self.sagemaker_client.delete_endpoint_config(EndpointConfigName=endpoint_config_name)
+
+    def delete_model(self, model_name):
+        """Delete an Amazon SageMaker ``Model``.
+
+        Args:
+            model_name (str): Name of the Amazon SageMaker model to delete.
+
+        Raises: ValueError if model is not deployed yet.
+
+        """
+        try:
+            self.sagemaker_client.describe_model(ModelName=model_name)
```

Comment:
My mistake, I think keeping the none check within the model class makes sense because that class handles the deploy, while in the session class it should just call the API on it's own.

This is a little dangerous as the corresponding exception message is catering towards the model not being deployed, when in fact it can exist, however not have a corresponding endpoint.

I think having this method just do sagemaker_client.deleteModel() is fine. You shouldn't do the extra check here.